cpp/include/experimental/graph_view.hpp

```diff
@@ -170,22 +179,27 @@ class partition_t {
 
   std::tuple<vertex_t, vertex_t> get_matrix_partition_minor_range() const
   {
-    auto minor_first = vertex_partition_offsets_[col_comm_rank_ * row_comm_size_];
-    auto minor_last  = vertex_partition_offsets_[(col_comm_rank_ + 1) * row_comm_size_];
+    auto minor_first = get_matrix_partition_minor_first();
+    auto minor_last  = get_matrix_partition_minor_last();
 
     return std::make_tuple(minor_first, minor_last);
   }
 
   vertex_t get_matrix_partition_minor_first() const
   {
-    return vertex_partition_offsets_[col_comm_rank_ * row_comm_size_];
+    return hypergraph_partitioned_ ? vertex_partition_offsets_[col_comm_rank_ * row_comm_size_]
+                                   : vertex_partition_offsets_[row_comm_rank_ * col_comm_size_];
   }
 
   vertex_t get_matrix_partition_minor_last() const
   {
-    return vertex_partition_offsets_[(col_comm_rank_ + 1) * row_comm_size_];
+    return hypergraph_partitioned_
+             ? vertex_partition_offsets_[(col_comm_rank_ + 1) * row_comm_size_]
+             : vertex_partition_offsets_[(row_comm_rank_ + 1) * col_comm_size_];
   }
 
+  // FIXME: this function may be removed if we use the same partitioning strategy whether hypergraph
+  // partitioning is applied or not
```

Comment:
Yes, and there are already several branches for handling this in the implementation even though it is not used which is not ideal for controlling complexity in the long run.